components/automate-ui/src/app/entities/notification_rules/notification_rule.model.ts

```diff
@@ -67,6 +73,10 @@ export class NotificationRule implements RuleInterface {
     return Object.keys(this.AlertTypeLabels);
   }
 
+  public getTargetTypeKeys(): string[] {
+    return Object.keys(this.TargetTypeLabels);
+  }
+
   public toRequest(): Object {
```

Comment:
I noticed this now, I realize it is not in this branch but I think it was on your last branch, So just pointing it out to think about.  I believe you can get rid of all the commas in the target and rule.  Might also be nice to add a type to the target.. something like:

```
export interface TargetInterface {
  url: string;
  secret_id?: string;
  critical_controls_only?: boolean;
}
```
followed by something like

```
    let target: TargetInterface = {url: this.targetUrl};
    if (this.targetSecretId !== undefined && this.targetSecretId !== '' ) {
      target = {
        url: this.targetUrl,
        secret_id: this.targetSecretId,
        critical_controls_only: this.criticalControlsOnly
      };
    }
```

I haven't tested thoroughly, just pointing it out and seems to work.  I know that we hope to avoid using the type of `any` whenever possible.